### PR TITLE
Implement client for query suggestions API

### PR DESCRIPTION
**Pull request by k-yomo**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes    
| BC breaks?        | no     
| Related Issue     | #658 
| Need Doc update   | yes

## Describe your change
Implemented Query Suggestions API client excepts `get configuration status` and `get a log file`.
https://www.algolia.com/doc/rest-api/query-suggestions/#quick-reference

### CODE REVIEW

File: algolia/query_suggestions/client_config.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	Facets []map[string]interface{}
+	// Minimum number of hits (e.g., matching records in the source index) to generate a suggestions.
+	MinHits *int `json:"minHits"`
+	// Minimum number of required letters for a suggestion to remain.
+	MinLetters *int `json:"minLetters"`
+	// List of facet attributes used to generate Query Suggestions. The resulting suggestions are every combination of the facets in the nested list
+	Generate [][]string `json:"generate,omitempty"`
+	// TODO: Figure out what type we should use for external.
+	// List of external indices to use to generate custom Query Suggestions.
+	// External []string `json:"external,omitempty"`
```
Comment (k-yomo) [inline algolia/query_suggestions/client_config.go:35]: Please check what type is appropriate for the `external`.

**Comment by k-yomo** [comment]
@VladislavFitz @Nagriar @DevinCodes 
Could someone review this PR or assign a reviewer?🙏

I would like to have a resource for query suggestions in the below terraform provider.
https://github.com/k-yomo/terraform-provider-algolia

**Comment by DevinCodes** [comment]
Assigning you @VladislavFitz, I don't know enough about Go to be of help here...

File: algolia/query_suggestions/client_config.go (hunk 1 above)
Comment (VladislavFitz) [inline algolia/query_suggestions/client_config.go:35]:
```suggestion
	External []string `json:"external,omitempty"`
```
This is the list of indices names, so the `[]string` type is a good one.

File: algolia/query_suggestions/client_config.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package query_suggestions
+
+import (
+	"fmt"
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/call"
+	"net/http"
+)
+
+type IndexConfig struct {
```
Comment (VladislavFitz) [inline algolia/query_suggestions/client_config.go:9]:
```suggestion
type IndexConfiguration struct {
```
We encourage to use the full term names for models. 
It's still appropriate to use the `Config` in the methods names as they match the actual API endpoints paths.

**Review by VladislavFitz — COMMENTED** [review]
Hi @k-yomo,
Thank you for such a great contribution and sorry for the delay.
I left two comments to consider. 
At the moment we have a CI issue so it doesn't work well with the  community PRs. We are working on it. Once it's fixed I'll merge your PRs and will release a new version of the client 👌

File: algolia/query_suggestions/client_config.go (hunk 1 above)
Comment (k-yomo) [reply algolia/query_suggestions/client_config.go]: Thank you for your confirmation! Fixed to use `[]string`

File: algolia/query_suggestions/client_config.go (hunk 2 above)
Comment (k-yomo) [reply algolia/query_suggestions/client_config.go]: I see, I replaced it with `IndexConfiguration`.

**Comment by k-yomo** [comment]
Hi @VladislavFitz 
Thank you for the review!!
I fixed things related to your comments🙌

> At the moment we have a CI issue so it doesn't work well with the community PRs. We are working on it. Once it's fixed I'll merge your PRs and will release a new version of the client 👌

I see, I'm looking forward to it!

**Review by VladislavFitz — APPROVED** [review]

**Pull request merged**